<?xml version="1.0" ?>
<sdf version="1.8">
    <world name="custom_world">
        <physics name="default_physics" type="ignition">
            <max_step_size>0.001</max_step_size>
            <real_time_factor>1</real_time_factor>
        </physics>
        <plugin
            filename="gz-sim-physics-system"
            name="gz::sim::systems::Physics">
        </plugin>
        <plugin
            filename="gz-sim-user-commands-system"
            name="gz::sim::systems::UserCommands">
        </plugin>
        <plugin
            filename="gz-sim-scene-broadcaster-system"
            name="gz::sim::systems::SceneBroadcaster">
        </plugin>
        <plugin
            filename="libignition-gazebo-sensors-system.so"
            name="ignition::gazebo::systems::Sensors">
            <render_engine>ogre2</render_engine>
        </plugin>
        <plugin filename="libignition-gazebo-imu-system.so"
            name="ignition::gazebo::systems::Imu">
        </plugin>
        <light type="directional" name="sun">
            <cast_shadows>true</cast_shadows>
            <pose>0 0 10 0 0 0</pose>
            <diffuse>0.8 0.8 0.8 1</diffuse>
            <specular>0.2 0.2 0.2 1</specular>
            <attenuation>
                <range>1000</range>
                <constant>0.9</constant>
                <linear>0.01</linear>
                <quadratic>0.001</quadratic>
            </attenuation>
            <direction>-0.5 0.1 -0.9</direction>
        </light>
        <model name="ground_plane">
            <static>true</static>
            <link name="link">
                <collision name="collision">
                    <geometry>
                        <plane>
                            <normal>0 0 1</normal>
                        </plane>
                    </geometry>
                </collision>
                <visual name="visual">
                    <geometry>
                        <plane>
                            <normal>0 0 1</normal>
                            <size>20 20</size>
                        </plane>
                    </geometry>
                    <material>
                        <ambient>0.8 0.8 0.8 1</ambient>
                        <diffuse>0.8 0.8 0.8 1</diffuse>
                        <specular>0.8 0.8 0.8 1</specular>
                    </material>
                </visual>
            </link>
        </model>

        <model name="wall1">
            <static>true</static>
            <pose>5 0 1 0 0 0</pose>
            <link name="link">
                <collision name="collision">
                    <geometry>
                        <box>
                            <size>0.5 10 2</size>
                        </box>
                    </geometry>
                </collision>
                <visual name="visual">
                    <geometry>
                        <box>
                            <size>0.5 10 2</size>
                        </box>
                    </geometry>
                    <material>
                        <ambient>0.7 0.2 0.2 1</ambient>
                        <diffuse>0.7 0.2 0.2 1</diffuse>
                        <specular>0.7 0.2 0.2 1</specular>
                    </material>
                </visual>
            </link>
        </model>

        <model name="wall2">
            <static>true</static>
            <pose>0 2 1 0 0 0</pose>
            <link name="link">
                <collision name="collision">
                    <geometry>
                        <box>
                            <size>4 0.5 2</size>
                        </box>
                    </geometry>
                </collision>
                <visual name="visual">
                    <geometry>
                        <box>
                            <size>4 0.5 2</size>
                        </box>
                    </geometry>
                    <material>
                        <ambient>0.7 0.2 0.2 1</ambient>
                        <diffuse>0.7 0.2 0.2 1</diffuse>
                        <specular>0.7 0.2 0.2 1</specular>
                    </material>
                </visual>
            </link>
        </model>

        <model name="wall3">
            <static>true</static>
            <pose>8 2 1 0 0 0</pose>
            <link name="link">
                <collision name="collision">
                    <geometry>
                        <box>
                            <size>1 10 2</size>
                        </box>
                    </geometry>
                </collision>
                <visual name="visual">
                    <geometry>
                        <box>
                            <size>1 10 2</size>
                        </box>
                    </geometry>
                    <material>
                        <ambient>0.7 0.2 0.2 1</ambient>
                        <diffuse>0.7 0.2 0.2 1</diffuse>
                        <specular>0.7 0.2 0.2 1</specular>
                    </material>
                </visual>
            </link>
        </model>

        <model name="wall4">
            <static>true</static>
            <pose>1 -5 1 0 0 0</pose>
            <link name="link">
                <collision name="collision">
                    <geometry>
                        <box>
                            <size>0.5 10 2</size>
                        </box>
                    </geometry>
                </collision>
                <visual name="visual">
                    <geometry>
                        <box>
                            <size>0.5 10 2</size>
                        </box>
                    </geometry>
                    <material>
                        <ambient>0.7 0.2 0.2 1</ambient>
                        <diffuse>0.7 0.2 0.2 1</diffuse>
                        <specular>0.7 0.2 0.2 1</specular>
                    </material>
                </visual>
            </link>
        </model>

        <model name="wall5">
            <static>true</static>
            <pose>-4 2.5 1 0 0 0</pose>
            <link name="link">
                <collision name="collision">
                    <geometry>
                        <box>
                            <size>0.5 15 2</size>
                        </box>
                    </geometry>
                </collision>
                <visual name="visual">
                    <geometry>
                        <box>
                            <size>0.5 15 2</size>
                        </box>
                    </geometry>
                    <material>
                        <ambient>0.7 0.2 0.2 1</ambient>
                        <diffuse>0.7 0.2 0.2 1</diffuse>
                        <specular>0.7 0.2 0.2 1</specular>
                    </material>
                </visual>
            </link>
        </model>

        <model name="wall6">
            <static>true</static>
            <pose>-8 3 1 0 0 0</pose>
            <link name="link">
                <collision name="collision">
                    <geometry>
                        <box>
                            <size>4 0.5 2</size>
                        </box>
                    </geometry>
                </collision>
                <visual name="visual">
                    <geometry>
                        <box>
                            <size>4 0.5 2</size>
                        </box>
                    </geometry>
                    <material>
                        <ambient>0.7 0.2 0.2 1</ambient>
                        <diffuse>0.7 0.2 0.2 1</diffuse>
                        <specular>0.7 0.2 0.2 1</specular>
                    </material>
                </visual>
            </link>
        </model>

        <model name="box1">
            <static>true</static>
            <pose>-2 -3 1 0 0 0</pose>
            <link name="link">
                <collision name="collision">
                    <geometry>
                        <box>
                            <size>1 1 2</size>
                        </box>
                    </geometry>
                </collision>
                <visual name="visual">
                    <geometry>
                        <box>
                            <size>1 1 2</size>
                        </box>
                    </geometry>
                    <material>
                        <ambient>0.7 0.2 0.2 1</ambient>
                        <diffuse>0.7 0.2 0.2 1</diffuse>
                        <specular>0.7 0.2 0.2 1</specular>
                    </material>
                </visual>
            </link>
        </model>

        <model name="box2">
            <static>true</static>
            <pose>-5 -3 1 0 0 0</pose>
            <link name="link">
                <collision name="collision">
                    <geometry>
                        <box>
                            <size>2.5 1 2</size>
                        </box>
                    </geometry>
                </collision>
                <visual name="visual">
                    <geometry>
                        <box>
                            <size>2.5 1 2</size>
                        </box>
                    </geometry>
                    <material>
                        <ambient>0.7 0.2 0.2 1</ambient>
                        <diffuse>0.7 0.2 0.2 1</diffuse>
                        <specular>0.7 0.2 0.2 1</specular>
                    </material>
                </visual>
            </link>
        </model>

            <include>
                <uri>file://./my_robot_3.sdf</uri>
                <pose>-10 10 0 0 0 0</pose>
            </include>
        
    </world>
</sdf>
